src/Http/Routing/src/ShortCircuit/RouteShortCircuitEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing;
+
+/// <summary>
+/// Provides extension methods for <see cref="IEndpointRouteBuilder"/> to add short circuited endpoints.
+/// </summary>
+public static class RouteShortCircuitEndpointRouteBuilderExtensions
+{
+    private static readonly RequestDelegate _shortCircuitDelegate = (context) => Task.CompletedTask;
+    /// <summary>
+    /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP GET requests
+    /// for the specified prefixes.
+    /// </summary>
+    ///<param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+    /// <param name="statusCode">The status code to set in the response.</param>
+    /// <param name="routePrefixes">An array of route prefixes to be short circuited.</param>
+    /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+    public static IEndpointConventionBuilder MapShortCircuit(this IEndpointRouteBuilder builder, int statusCode, params string[] routePrefixes)
+    {
+        var group = builder.MapGroup("");
+        foreach (var routePrefix in routePrefixes)
+        {
+            string route;
+            if (routePrefix.EndsWith("/", StringComparison.OrdinalIgnoreCase))
+            {
+                route = $"{routePrefix}{{**catchall}}";
+            }
+            else
+            {
+                route = $"{routePrefix}/{{**catchall}}";
+            }
+            group.Map(routePrefix, _shortCircuitDelegate).ShortCircuit(statusCode);
```

Comment:
@javiercn suggested giving this a low priority Order to avoid conflicts with other routes in the app.

Reply:
I couldn't figure out an easy way to set the Order. Is there an API to do that?

Reply:
```
group.Map(routePrefix, _shortCircuitDelegate).Add(endpoint => ((RouteEndpoint)endpoint).Order = value)
```

Reply:
Ah! I have to cast it! thanks @javiercn